massa-execution-worker/src/context.rs

```diff
@@ -320,6 +331,7 @@ impl ExecutionContext {
             gas_remaining_before_subexecution: self.gas_remaining_before_subexecution,
             recursion_counter: self.recursion_counter,
             user_event_count_in_current_exec: self.user_event_count_in_current_exec,
+            transfer_history: self.transfers_history.read().clone(),
```

Comment:
Potentially very heavy...

here is an idea for optimizing, a bit like we did for async messages, since the elements are emitted in order:
- get the item count at snapshot
- in case of snapshot revert, simply truncate the transfer history vector to the saved item count

Reply:
done